source/particle/integrator/euler.cc

```diff
@@ -49,7 +55,12 @@ namespace aspect
              it != end_particle; ++it, ++old_velocity)
           {
             const Point<dim> loc = it->get_location();
-            it->set_location(loc + dt * (*old_velocity));
+            Point<dim> new_location = loc + dt * (*old_velocity);
+
+            if (has_periodic_boundary)
+              this->get_geometry_model().adjust_positions_for_periodicity(new_location);
```

Comment:
This is the place where I mention that you use the `has_periodic_boundary` variable. I think you might as well inline the check for periodicity in the `if` condition.